src/metabase/util/yaml.clj

```diff
@@ -1,27 +1,63 @@
 (ns metabase.util.yaml
+  "Convenience functions for parsing and generating YAML."
   (:refer-clojure :exclude [load])
   (:require
+   [clj-yaml.core :as yaml]
+   [clojure.java.io :as io]
    [clojure.string :as str]
    [metabase.util :as u]
+   [metabase.util.date-2 :as u.date]
    [metabase.util.files :as u.files]
    [metabase.util.i18n :refer [trs]]
-   [metabase.util.log :as log]
-   [yaml.core :as yaml])
+   [metabase.util.log :as log])
   (:import
-   (java.nio.file Files Path)))
+   (java.nio.file Files Path)
+   (java.time.temporal Temporal)))
 
 (set! *warn-on-reflection* true)
 
+(defn- vectorized
+  "Returns x with lazy seqs converted to vectors wherever they appear in the data structure."
+  [x]
+  (cond
+    (map? x)        (zipmap (keys x) (map vectorized (vals x)))
+    (sequential? x) (mapv vectorized x)
+    :else           x))
+
+(defn- yamlize
+  "Returns x transformed for YAML output, converting dates to strings using a standard format."
+  [x]
+  (cond
+    (instance? Temporal x) (u.date/format x)
+    (map? x)               (zipmap (keys x) (map yamlize (vals x)))
+    (sequential? x)        (map yamlize x)
+    :else                  x))
+
+(defn from-file
```

Comment:
So this namespace is a drop-in replacement for the old `yaml.core`, because `clj-yaml` doesn't have `from-file`.